src/Functional/HArrayFilterWithKey.php

```diff
@@ -20,6 +20,16 @@ public function __construct(array $arr)
      */
     public function filter(callable $func)
     {
+        if (!defined('ARRAY_FILTER_USE_KEY')) {
+            $return = [];
+            foreach ($this->arr as $k => $v) {
+                if (call_user_func($func, $k)) {
+                    $return[$k] = $v;
+                }
+            }
+            return $return;
+        }
+
```

Comment:
Not totally thrilled with the polyfill implementations but tests pass! If there are other ways to implement this I'm open.